test/Npgsql.Tests/Types/NumericTests.cs

```diff
@@ -161,6 +162,53 @@ public async Task ReadOverflowIsSafe()
             }
         }
 
+        [Test, Description("Tests the GetSqlDecimal for the Read cases")]
+        [TestCaseSource(nameof(ReadWriteCases))]
+        public async Task ReadSqlDecimal(string query, decimal expected) {
+            using (var conn = await OpenConnectionAsync())
+            using (var cmd = new NpgsqlCommand("SELECT " + query, conn)) {
+                using var reader = await cmd.ExecuteReaderAsync();
+                reader.Read();
+
+                var result = reader.GetSqlDecimal(0);
+                Assert.That(
+                    decimal.GetBits(result.Value),
+                    Is.EqualTo(decimal.GetBits(expected)));
+            }
+        }
+
+        [Test, Description("Tests when GetSqlDecimal for Numeric values that don't fit in a System.Decimal and they have decimal digits which can be rounded to fit")]
+        public async Task ReadSqlDecimalOverflow() {
+
+            using (var conn = OpenConnection()) {
+                using (var cmd = new NpgsqlCommand()) {
+                    cmd.Connection = conn;
+
+                    var readCases = new[] {
+                                            new object[]{ "1000000000000000000000.123456448", 1000000000000000000000.123456448m , 29, 7 },
```

Comment:
It's probably a good idea to add testing also in the Read tests above, this way we're sure that you're code also works for all the cases already tested with .NET decimal.